Automated demo sequence for testing

awwad opened the pull request:
### Summary
Primarily in the interests of facilitating complete testing before merging things, here's a draft of a script that runs the current console demo.

### Details
Running python demo/demo_automation.py executes all the
commands in the console demo, along with notes about expected
output.

The primary purpose is for a fairly quick additional round of testing,
to make sure that any changes made don't break the demo. Currently, **checking
to make sure everything ran correctly requires that you scan up through the
output and check the yellow instructions against the preceding banners.**
More code is required to automate those checks, allowing this to be run
as a test script, but this is helpful for now.

To run the demo this way, run the following from the main uptane directory
(which contains, for example, setup.py).
  python demo/demo_automation.py

If 10 instructions in yellow text appear (e.g. "The preceding banner should
be: NO UPDATE") and the banners indicated match, then the demo has run
correctly.

awwad committed:
DEMO: First draft of automated demo sequence (for testing)

Running python -i demo/demo_automation.py executes all the
commands in the console demo, along with notes about expected
output. I'll make this into a more useful test script next.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

awwad committed:
DOC: Add more doc for single-command demo run

to clarify purpose of the module and how to use it.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

awwad committed:
DOC: Simplify single-command demo instructions

by removing suggestion that an interactive Python session be used.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

awwad committed:
Rename single-run demo module for clarity

from demo_automation to demo_single_command_run

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

awwad committed:
DOC: correct a demo module name in docstring

demo_single_command_run
(Kept as a separate commit from the git mv command for
better diff viewing.)

Also adds a warning that this script does not take the place
of the test suite for the reference implementation.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

vladimir-v-diaz commented on demo/demo_automation.py:
```diff
@@ -13,11 +13,7 @@
-  That starts the demo in an interactive mode (with a prompt from which
-  you can manipulate commands for the demonstrations). The demo is run in full
-  before any commands can be entered.
+    python demo/demo_automation.py
```
Don't forget to update the pull request summary to match the change.


What's current in the summary:
```
Running python -i demo/demo_automation.py executes all the
commands in the console demo, along with notes about expected
output. I'll make this into a more useful test script next.
```

awwad replied:
Surely will, thanks :)

awwad committed:
Demo: clean up after single-command demo run

in a manner that will work for both interactive Python sessions
and script runs (atexit).

Deletes director/ and imagerepo/ as well as firmware.img, which
are created when the demo is run.

Also removes readline/rlcompleter imports and use, since the
script may not be run as an interactive session.

Also removes an old commented-out import.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>
